xds/src/main/java/io/grpc/xds/XdsClientWrapperForServerSds.java

```diff
@@ -239,6 +243,46 @@ private static SslContextProviderSupplier getSslContextProviderSupplier(
     return listener.getDefaultFilterChain().getSslContextProviderSupplier();
   }
 
+  // reject if filer-chain-match has non-empty application_protocols
+  private static List<FilterChain> filterOnApplicationProtocols(List<FilterChain> filterChains) {
+    ArrayList<FilterChain> filtered = new ArrayList<>(filterChains.size());
+    for (FilterChain filterChain : filterChains) {
+      FilterChainMatch filterChainMatch = filterChain.getFilterChainMatch();
+
+      if (filterChainMatch.getApplicationProtocols().isEmpty()) {
+        filtered.add(filterChain);
+      }
+    }
+    return filtered;
+  }
+
+  // reject if filer-chain-match has non-empty transport protocol other than "raw_buffer"
+  private static List<FilterChain> filterOnTransportProtocol(List<FilterChain> filterChains) {
+    ArrayList<FilterChain> filtered = new ArrayList<>(filterChains.size());
+    for (FilterChain filterChain : filterChains) {
+      FilterChainMatch filterChainMatch = filterChain.getFilterChainMatch();
+
+      String transportProtocol = filterChainMatch.getTransportProtocol();
+      if ( Strings.isNullOrEmpty(transportProtocol) || "raw_buffer".equals(transportProtocol)) {
```

Comment:
The `transportProtocol` can never be null for `EnvoyServerDataProto.FilterChainMatch`, as it's directly gotten from the proto.